controller/RestTestCenterUsers.php

```diff
@@ -206,25 +240,42 @@ public function get()
 
     /**
      * @return \oat\oatbox\user\User
-     * @throws \common_exception_BadRequest
      * @throws \common_exception_Error
-     * @throws \common_exception_MissingParameter
+     * @throws \common_exception_NotFound
+     * @throws \common_exception_RestApi
      */
     private function getUserFromRequest()
     {
-        $userUri = $this->getParameterFromRequest(self::PARAMETER_USER_URI);
-        return $this->getUserService()->getUserById($userUri);
+        try {
+            $userUri = $this->getParameterFromRequest(self::PARAMETER_USER_URI);
+        } catch (\common_exception_MissingParameter $e) {
+            throw new \common_exception_RestApi(__('Missed required parameter: `%s`', self::PARAMETER_USER_URI));
+        }
+        $user = $this->getUserService()->getUserById($userUri);
+        if (!$user || !$this->getResource($user->getIdentifier())->exists()) {
+            throw new \common_exception_NotFound(__('User `%s` does not exist.', $userUri));
+        }
+
+        return $user;
     }
 
     /**
      * @return \core_kernel_classes_Resource
-     * @throws \common_exception_MissingParameter
      * @throws \common_exception_NotFound
+     * @throws \common_exception_RestApi
      */
     private function getRoleFromRequest()
     {
-        $roleUri = $this->getParameterFromRequest(self::PARAMETER_USER_ROLE);
-        return $this->getAndCheckResource($roleUri, 'http://www.tao.lu/Ontologies/generis.rdf#UserRole');
+        $roleUri = '';
+        try {
+            $roleUri = $this->getParameterFromRequest(self::PARAMETER_USER_ROLE);
+
+            return $this->getAndCheckResource($roleUri, 'http://www.tao.lu/Ontologies/generis.rdf#UserRole');
```

Comment:
Any reason you preferred hardcoded value instead of using constant UserRdf::PROPERTY_ROLES ?

Reply:
The constant `UserRdf::PROPERTY_ROLES` refers to a User entity field, assigned roles to the user. While in this case the code checks if the provided role is valid value for entity `UserRole`.
I didn't find a constant for class `http://www.tao.lu/Ontologies/generis.rdf#UserRole`. Would you suggest to add it to `GenerisRdf` or `UserRdf`?